lua/nvim-treesitter/query_predicates.lua

```diff
@@ -71,4 +72,30 @@ M['has_ancestor?'] = function(query, match, pred)
   return false
 end
 
+M['adjacent?'] = function(query, match, pred)
+  if #pred < 3 then error("adjacent? must have at least two arguments!") end
+  local node = get_node(query, match, pred[2])
+  if not node then return true end
+
+  local adjacent_types = {unpack(pred, 3)}
+  local adjacent_node = ts_utils.get_next_node(node, false, false)
```

Comment:
You can leave away the optional parameters. Maybe we should remove them altogether from the function since additional crazyness for node movement is no longer required (node movement is also with `false,false` better.